mirgecom/mpi.py

```diff
@@ -0,0 +1,65 @@
+""":mod:`mirgecom.mpi` contains MPI helper functionality."""
+
+__copyright__ = """
+Copyright (C) 2020 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+
+from functools import wraps
+import os
+import sys
+
+
+def mpi_entry_point(func):
+    """
+    Designate a function as the "main" function for MPI.
+
+    In other words, declares that all MPI code that will be executed on the current
+    process is contained within *func*. Calls `MPI_Init()`/`MPI_Init_thread()` and
+    sets up a hook to call `MPI_Finalize()` on exit.
+    """
+    @wraps(func)
+    def wrapped_func(*args, **kwargs):
+        if "mpi4py.MPI" in sys.modules:
+            raise RuntimeError("mpi4py.MPI imported before designated MPI entry "
+                        "point.")
```

Comment:
Provide advice on what to do to fix this.

Reply:
I added a little more, but now I wonder if it basically amounts to saying "Don't do that.". Were you thinking it should say something more specific? Is there a way to check where a module was imported, maybe?